src/resolution_functions/instrument.py

```diff
@@ -643,8 +667,8 @@ def available_models_and_configurations(self) -> dict[str, list[str]]:
         models_and_configurations
             All models and all their configurations.
         """
-        return {model_name: list(model['configurations'].keys())
-                for model_name, model in self._models.items()}
+        return {model_name: list(self._resolve_model(model_name)['configurations'].keys())
+                for model_name in self._models.keys()}
```

Comment:
With the current implementation, this is how this function would look like (the question whether it should return *all* models as it currently does, or only the non-versioned ones is separate), but I think the loop over `keys()` and then using the key to index into the dictionary is not the most elegant.

Perhaps we should create a new class (a `dict` subclass maybe) for storing the models dictionary that could handle the aliases nicer?

Reply:
I think the `.keys()` are redundant because `list` and `in` work with keys by default.

Reply:
Given that Python handles things as pointers, dictionaries can just point to the same entity and be aliases.

Reply:
We have decided to turn this (and similar) property into ones that do not output a dictionary but a nicely formatted string - this way, the methods (such as `possible_configurations_for_model`) will be meant for API access by codes (when a type is required), and these properties will be meant for users trying to get information in REPL etc. This will reduce the need for special stuff with `self._models` (and in fact means we'll want to keep the distinction between specific models and aliases so that we can use it for formatting)

Reply:
This has now been implemented